okio-wasifilesystem/src/wasmMain/kotlin/okio/internal/Wasi.kt

```diff
@@ -42,17 +42,33 @@ internal fun Pointer.readString(byteCount: Int): String {
 
 private fun Pointer.readByteArray(byteCount: Int): ByteArray {
   val result = ByteArray(byteCount)
-  for (i in 0 until byteCount) {
-    result[i] = (this + i).loadByte()
-  }
+  read(result, 0, byteCount)
   return result
 }
 
+internal fun Pointer.read(
+  sink: ByteArray,
+  offset: Int,
+  count: Int,
+): ByteArray {
+  for (i in 0 until count) {
+    sink[offset + i] = (this + i).loadByte()
+  }
+  return sink
+}
+
 internal fun MemoryAllocator.write(
   string: String,
 ): Pair<Pointer, size> {
   val bytes = string.encodeToByteArray()
-  return write(bytes) to bytes.size
+  val result = allocate(bytes.size + 1)
```

Comment:
This was a bug before! Linux gets really wrecked unless we add a trailing \0 when writing strings

Reply:
(even though we also provide the string length elsewhere)

Reply:
FYI @sdeleuze I needed a trailing `\0` on strings for NodeJS to not crash on Linux.